core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -196,6 +197,24 @@ class HadoopRDD[K, V](
     newInputFormat
   }
 
+  private val UNSPLITTABLE_FILE_SIZE_LOG_THRESHOLD = 1024 * 1024 * 1024
+
+  @transient private lazy val compressionCodecs = new CompressionCodecFactory(getJobConf())
+
+  private def checkAndLogUnsplittableLargeFile(split: InputSplit): Unit = {
+    if (split.isInstanceOf[FileSplit]) {
+      val fileSplit = split.asInstanceOf[FileSplit]
+      val path = fileSplit.getPath
+      val codec = compressionCodecs.getCodec(path)
+      if (codec != null && !codec.isInstanceOf[SplittableCompressionCodec]) {
```

Comment:
Here I duplicate `org.apache.hadoop.mapred.TextInputFormat.isSplitable` code logic because it is a protected method.